ksp_plugin_test/plugin_compatibility_test.cpp

```diff
@@ -113,17 +126,109 @@ class PluginCompatibilityTest : public testing::Test {
                                       preferred_compressor,
                                       preferred_encoder);
   }
+
+  static void CheckSaveCompatibility(std::filesystem::path const& filename,
+                              std::string_view const compressor,
+                              std::string_view const encoder) {
+    // Read a plugin from the given file.
+    auto plugin = ReadPluginFromFile(filename, compressor, encoder);
+
+    WriteAndReadBack(std::move(plugin));
+  }
 };
 
 TEST_F(PluginCompatibilityTest, PreCartan) {
   // This space for rent.
 }
 
 TEST_F(PluginCompatibilityTest, PreCohen) {
+  CaptureStderr();
   CheckSaveCompatibility(
-      SOLUTION_DIR / "ksp_plugin_test" / "pre_cohen.proto.hex",
+      SOLUTION_DIR / "ksp_plugin_test" / "saves" / "3039.proto.hex",
       /*compressor=*/"",
       /*decoder=*/"hexadecimal");
+  EXPECT_THAT(
+      GetCapturedStderr(),
+      AllOf(
+          HasSubstr(
+              "pre-Cohen ContinuousTrajectory"),  // Regression test for #3039.
```

Comment:
Not convinced that the comments here add value.

Reply:
At least this one seems useful, since this this is what #3039 was about, and it’s not particularly obvious from the phrase « pre-Cohen ContinuousTrajectory ».